hal_st/stm32fxxx/AnalogToDigitalPinStm.hpp

```diff
@@ -12,10 +12,11 @@ namespace hal
     class AdcStm;
 
     class AnalogToDigitalPinImplStm
-        : protected AnalogToDigitalPinImplBase<uint16_t>
+        : public AnalogToDigitalPinImplBase<uint16_t>
     {
     public:
-        AnalogToDigitalPinImplStm(hal::GpioPinStm& pin, AdcStm& adc);
+        explicit AnalogToDigitalPinImplStm(hal::GpioPinStm& pin, AdcStm& adc);
+        virtual ~AnalogToDigitalPinImplStm() = default;
```

Comment:
Why does this destructor need to be virtual?

Reply:
It does not need, indeed. Destructor removed.